### ISSUE TITLE
Document middleware phases

### ISSUE THREAD

**Issue by bajtos** [strongloop/loopback#863]
Update the middleware documentation page in Confluence and describe the new features related to middleware phases:

  http://docs.strongloop.com/display/LB/Defining+middleware

A draft of the new content:

  https://gist.github.com/bajtos/e7eaba736ff096916b71

Before we can publish the new content, the following tasks must be finished:
- [x] Land https://github.com/strongloop/loopback-boot/pull/70
- [x] Release a new minor version of loopback-boot (v2.4.0)

Nice to have: 
- [ ] https://github.com/strongloop/loopback-workspace/issues/167
